lib/draper/decorator.rb

```diff
@@ -239,22 +239,23 @@ def self.alias_object_to_object_class_name
     end
 
     def self.object_class_name
-      raise NameError if name.nil? || name.demodulize !~ /.+Decorator$/
+      return nil if name.nil? || name.demodulize !~ /.+Decorator$/
       name.chomp("Decorator")
     end
 
     def self.inferred_object_class
       name = object_class_name
-      name.constantize
-    rescue NameError => error
-      raise if name && !error.missing_name?(name)
-      raise Draper::UninferrableObjectError.new(self)
+      name_constant = name && name.safe_constantize
+      raise Draper::UninferrableObjectError.new(self) if name_constant.nil?
+
+      name_constant
     end
 
     def self.collection_decorator_name
-      plural = object_class_name.pluralize
-      raise NameError if plural == object_class_name
-      "#{plural}Decorator"
+      singular = object_class_name
+      plural = singular && singular.pluralize
+
+      plural == singular ? nil : "#{plural}Decorator"
```

Comment:
While the ternary statement works just fine, returning `nil` is the default behavior. So it might read better to say:

```ruby
singular = object_class_name
plural = singular&.pluralize
"#{plural}Decorator" unless plural == singular
```

What do you think?

Reply:
I feel that the ternary statement can make me more easily realize the 2 possible return types: `nil` and `"#{plural}Decorator"`, but I don't have strong opinion on it, so it's fine to change

```ruby
"#{plural}Decorator" unless plural == singular
```

Reply:
My opinion isn't strong either. I see the advantages of the ternary as well. Completely up to you.